specs/slider/slider.md

```diff
@@ -0,0 +1,181 @@
+
+### Background
+
+Just styling `<input type="range"/>` and using `<datalist>` for the tickmarks was considered but the standards aren't supported enough to allow for the level customization we want to provide.
+
+** see [MDN Docs](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/range)
+
+### Use Cases
+
+- *A customer using the component on a web page.*
+On a web page the customer drags the slider from min to max values ultimately setting the value of the component to a value constrained by step and min and max values.
+
+- *A developer building an app with the component and interacting through HTML/CSS/JavaScript.*
+A developer can inject labels on the slider into the default slot, if they use <slider-label> elements those will be styled a specific way internally and any part of that style can be overridden. Styles and items intending to be placed in the default slot can be replaced as needed.
+
+- *A designer customizing the component.*
+A designer can override any internal styling applied to the slotted labels, the background track, the slider track and the thumb(s) if appropriate.
+  
+### Features
+- Implement your own lower and upper thumb UI, default provided
+- Implement your own background and slider track UI, default provided
+- Handle the `slider`'s `onValueChange` callback to be notified when the value has changed.
+- Customize the tick marks along the track
+- Include slider in form based operations
+
+### Prior Art/Examples
+- [FAST-DNA (React)](https://explore.fast.design/components/slider)
+- [Material UI](https://material-ui.com/components/slider/)
+
+---
+
+### API
+
+*Component Name*:
+- `fast-slider`
+
+*Props/Attrs*
+- `min` - minimum allowed value for the slider
+- `max` - maximum allowed value for the slider
+- `step` - limits the values of the `slider` to increments of the `step` value added to the minimum value of the 
+`slider`'s total range.  The default value is 1. The minimum and maximum values of a `slider`'s range are always valid results regardless of the `step` prop. The `step` prop is used as the value for incrementing the thumb by pressing the arrow keys.
+ - `value` - Allows authors to specify the initial selected range of the `slider`.  It defaults to a (step constrained) value at the midpoint on the `slider`'s total range. 
+
+*Events*
+- `change` - raise the change event for external parties to be informed of the value change.
+
+*Slots*
+- `background-track` - background area encompassing the track and labels
+- `slider-track` - the horizontal or vertical track along which the thumb slides
+- `thumb` - the control the user drags along the track to change the value of the slider
+
+### Anatomy and Appearance
+**Structure:**
+
+```html
+  <div
+    role="slider"
+  >
+    <input type="range" 
+      id="${x => x.id}"
+      min="${x => x.min}"
+      max="${x => x.max}"
+      step="${x => x.step}"
+      value="${x => x.value}"
+    >

```

Comment:
Do you plan on using form association here? The input being rendered here is throwing me off. I think you'd want the input here if you planned on delegating focus to it; however, since we can't style the element well enough, I think that's an unlikely scenario. I think we'll want this to be form associated instead. You can see an example of the differences by comparing the checkbox web component (form association) to the text field on I have in [review](https://github.com/microsoft/fast-dna/pull/2785) (focus delegation).

Reply:
@EisenbergEffect or @nicholasrice can correct me if I'm wrong here, btw :)

Reply:
Yes, this would use form-association.

Reply:
Yea, I've been building the component at the same time and it's using Form association and working. So I removed the input from the template in this spec and the actual code.